traffic_ops/traffic_ops_golang/deliveryservice/acme.go

```diff
@@ -257,43 +291,76 @@ func GenerateLetsEncryptCertificates(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	go GetAcmeCertificates(inf.Config, req, ctx, inf.User, inf.Vault)
+	asyncStatusId, errCode, userErr, sysErr := api.InsertAsyncStatus(inf.Tx.Tx, "ACME async job has started.")
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+	}
 
-	var alerts tc.Alerts
+	defer func() {
+		if err := recover(); err != nil {
+			db, dbErr := api.GetDB(ctx)
+			if dbErr != nil {
+				log.Errorf(*req.DeliveryService+": Error getting db for recover async update: %s", dbErr.Error())
+				api.HandleErr(w, r, nil, http.StatusInternalServerError, nil, fmt.Errorf("panic: (err: %v) stacktrace:\n%s\n", err, util.Stacktrace()))
+				return
+			}
 
-	alerts.AddAlerts(api.CreateDeprecationAlerts(util.StrPtr(API_ACME_GENERATE_LE)))
+			if asyncErr := api.UpdateAsyncStatus(db, api.AsyncFailed, "ACME renewal failed.", asyncStatusId, true); asyncErr != nil {
+				log.Errorf("updating async status for id %v: %v", asyncStatusId, asyncErr)
+			}
+			api.HandleErr(w, r, nil, http.StatusInternalServerError, nil, fmt.Errorf("panic: (err: %v) stacktrace:\n%s\n", err, util.Stacktrace()))
+			return
+		}
+	}()
```

Comment:
To test this recovery, I added a `panic` at the top of `GetAcmeCertificates` on line 331. Traffic Ops still crashed, and I never saw any of these log messages. So it appears that you must handle the panic from within the same goroutine that started it.